Parsing of params file with wildcards fails depending on ordering

### Generated by Generative AI

_No response_

### Operating System:

```Linux ubuntu 5.15.148-tegra```

### ROS version or commit hash:

humble

### RMW implementation (if applicable):

rmw_fastrtps_cpp

### RMW Configuration (if applicable):

_No response_

### Client library (if applicable):

rclpy

### 'ros2 doctor --report' output

<details><summary>ros2 doctor --report</summary>

```console
   NETWORK CONFIGURATION
inet         : 127.0.0.1
inet4        : ['127.0.0.1']
inet6        : ['::1']
netmask      : 255.0.0.0
device       : lo
flags        : 73<UP,LOOPBACK,RUNNING>
mtu          : 65536
inet         : 192.168.1.155
inet4        : ['192.168.1.155']
ether        : a8:e2:91:d9:b4:85
inet6        : ['fe80::5879:cba:3ff1:3038%wlP1p1s0']
netmask      : 255.255.255.0
device       : wlP1p1s0
flags        : 4163<MULTICAST,UP,RUNNING,BROADCAST>
mtu          : 1500
broadcast    : 192.168.1.255
inet         : 172.17.0.1
inet4        : ['172.17.0.1']
ether        : 0e:5d:8d:fc:eb:30
netmask      : 255.255.0.0
device       : docker0
flags        : 4099<MULTICAST,UP,BROADCAST>
mtu          : 1500
broadcast    : 172.17.255.255
ether        : 3c:6d:66:1e:c3:ea
device       : enP8p1s0
flags        : 4099<MULTICAST,UP,BROADCAST>
mtu          : 1500
ether        : 3e:c7:e7:79:f3:87
device       : l4tbr0
flags        : 4098<MULTICAST,BROADCAST>
mtu          : 1500
ether        : e6:b4:cb:34:8b:69
device       : usb0
flags        : 4099<MULTICAST,UP,BROADCAST>
mtu          : 1500
ether        : e6:b4:cb:34:8b:6b
device       : usb1
flags        : 4099<MULTICAST,UP,BROADCAST>
mtu          : 1500

   PACKAGE VERSIONS
rviz_common                               : latest=11.2.19, local=11.2.16
nav2_navfn_planner                        : latest=1.1.18, local=1.1.18
dwb_core                                  : latest=1.1.18, local=1.1.18
nav2_common                               : latest=1.1.18, local=1.1.18
ros2interface                             : latest=0.18.13, local=0.18.12
shape_msgs                                : latest=4.9.0, local=4.8.0
nav_2d_utils                              : latest=1.1.18, local=1.1.18
ament_cmake_target_dependencies           : latest=1.3.12, local=1.3.11
ament_cmake_core                          : latest=1.3.12, local=1.3.11
tracetools_image_pipeline                 : latest=3.0.8, local=3.0.7
lifecycle_msgs                            : latest=1.2.2, local=1.2.1
nav2_dwb_controller                       : latest=1.1.18, local=1.1.18
rqt_robot_monitor                         : latest=1.0.6, local=1.0.6
rosgraph_msgs                             : latest=1.2.2, local=1.2.1
navigation2                               : latest=1.1.18, local=1.1.18
nav2_planner                              : latest=1.1.18, local=1.1.18
console_bridge_vendor                     : latest=1.4.1, local=1.4.1
rosidl_cmake                              : latest=3.1.7, local=3.1.6
rosbag2_compression                       : latest=0.15.15, local=0.15.14
v4l2_camera                               : latest=0.6.2, local=0.6.2
foxglove_bridge                           : latest=0.8.5, local=0.8.5
tracetools                                : latest=4.1.1, local=4.1.1
pluginlib                                 : latest=5.1.0, local=5.1.0
rosbag2_interfaces                        : latest=0.15.15, local=0.15.14
nav2_collision_monitor                    : latest=1.1.18, local=1.1.18
launch_ros                                : latest=0.19.10, local=0.19.9
ament_cmake_export_include_directories    : latest=1.3.12, local=1.3.11
ament_cmake_cpplint                       : latest=0.12.14, local=0.12.12
ament_cmake_xmllint                       : latest=0.12.14, local=0.12.12
ros2run                                   : latest=0.18.13, local=0.18.12
qt_gui_cpp                                : latest=2.2.4, local=2.2.4
unique_identifier_msgs                    : latest=2.2.1, local=2.2.1
behaviortree_cpp_v3                       : latest=3.8.7, local=3.8.7
sros2                                     : latest=0.10.7, local=0.10.6
ament_package                             : latest=0.14.1, local=0.14.0
sqlite3_vendor                            : latest=0.15.15, local=0.15.14
map_msgs                                  : latest=2.1.0, local=2.1.0
nav2_velocity_smoother                    : latest=1.1.18, local=1.1.18
ament_lint_cmake                          : latest=0.12.14, local=0.12.12
camera_calibration                        : latest=3.0.8, local=3.0.7
yaml_cpp_vendor                           : latest=8.0.2, local=8.0.2
image_view                                : latest=3.0.8, local=3.0.7
compressed_image_transport                : latest=2.5.3, local=2.5.3
rcl_logging_interface                     : latest=2.3.1, local=2.3.1
angles                                    : latest=1.15.0, local=1.15.0
rmw_implementation                        : latest=2.8.5, local=2.8.4
rosidl_generator_py                       : latest=0.14.5, local=0.14.4
rmw_fastrtps_cpp                          : latest=6.2.8, local=6.2.7
ros_environment                           : latest=3.2.2, local=3.2.2
rqt_gui_cpp                               : latest=1.1.7, local=1.1.7
ros_base                                  : latest=0.10.0, local=0.10.0
camera_info_manager                       : latest=3.1.12, local=3.1.11
ament_pep257                              : latest=0.12.14, local=0.12.12
vision_opencv                             : latest=3.2.1, local=3.2.1
ros2cli                                   : latest=0.18.13, local=0.18.12
theora_image_transport                    : latest=2.5.3, local=2.5.3
sensor_msgs_py                            : latest=4.9.0, local=4.8.0
ament_cmake_export_interfaces             : latest=1.3.12, local=1.3.11
rmw_fastrtps_shared_cpp                   : latest=6.2.8, local=6.2.7
nav2_graceful_controller                  : latest=1.1.18, local=1.1.18
image_rotate                              : latest=3.0.8, local=3.0.7
slam_toolbox                              : latest=2.6.10, local=2.6.9
nav2_regulated_pure_pursuit_controller    : latest=1.1.18, local=1.1.18
rosidl_typesupport_interface              : latest=3.1.7, local=3.1.6
nav2_constrained_smoother                 : latest=1.1.18, local=1.1.18
ament_cmake_pep257                        : latest=0.12.14, local=0.12.12
demo_nodes_cpp                            : latest=0.20.5, local=0.20.5
rclpy                                     : latest=3.3.17, local=3.3.16
launch_testing_ament_cmake                : latest=1.0.10, local=1.0.8
vision_msgs                               : latest=4.1.1, local=4.1.1
libstatistics_collector                   : latest=1.3.4, local=1.3.4
rosidl_adapter                            : latest=3.1.7, local=3.1.6
tf2_geometry_msgs                         : latest=0.25.16, local=0.25.12
ament_lint_auto                           : latest=0.12.14, local=0.12.12
nav2_controller                           : latest=1.1.18, local=1.1.18
ament_cmake_libraries                     : latest=1.3.12, local=1.3.11
nav2_lifecycle_manager                    : latest=1.1.18, local=1.1.18
image_pipeline                            : latest=3.0.8, local=3.0.7
ros2cli_common_extensions                 : latest=0.1.1, local=0.1.1
dwb_critics                               : latest=1.1.18, local=1.1.18
osrf_pycommon                             : latest=2.1.6, local=2.1.6
diagnostic_updater                        : latest=4.0.6, local=4.0.3
libyaml_vendor                            : latest=1.2.2, local=1.2.2
python_cmake_module                       : latest=0.10.0, local=0.10.0
rosidl_runtime_py                         : latest=0.9.3, local=0.9.3
bond                                      : latest=4.1.2, local=4.1.2
nav2_map_server                           : latest=1.1.18, local=1.1.18
rmw_dds_common                            : latest=1.6.0, local=1.6.0
visualization_msgs                        : latest=4.9.0, local=4.8.0
rosidl_default_generators                 : latest=1.2.0, local=1.2.0
ros2topic                                 : latest=0.18.13, local=0.18.12
nav2_smoother                             : latest=1.1.18, local=1.1.18
ros2pkg                                   : latest=0.18.13, local=0.18.12
nav2_simple_commander                     : latest=1.1.18, local=1.1.18
image_geometry                            : latest=3.2.1, local=3.2.1
rviz_assimp_vendor                        : latest=11.2.19, local=11.2.16
message_filters                           : latest=4.3.8, local=4.3.7
rcl                                       : latest=5.3.10, local=5.3.9
rviz_ogre_vendor                          : latest=11.2.19, local=11.2.16
ament_flake8                              : latest=0.12.14, local=0.12.12
ament_cmake_cppcheck                      : latest=0.12.14, local=0.12.12
rclcpp                                    : latest=16.0.14, local=16.0.12
ament_cmake_python                        : latest=1.3.12, local=1.3.11
rosbag2_compression_zstd                  : latest=0.15.15, local=0.15.14
image_proc                                : latest=3.0.8, local=3.0.0
rosbridge_server                          : latest=2.0.1, local=2.0.0
urdf_parser_plugin                        : latest=2.6.1, local=2.6.1
ament_cmake_gtest                         : latest=1.3.12, local=1.3.11
ament_cmake_uncrustify                    : latest=0.12.14, local=0.12.12
ros2launch                                : latest=0.19.10, local=0.19.9
compressed_depth_image_transport          : latest=2.5.3, local=2.5.3
qt_gui                                    : latest=2.2.4, local=2.2.4
rcl_logging_spdlog                        : latest=2.3.1, local=2.3.1
rqt_image_view                            : latest=1.2.0, local=1.2.0
rcutils                                   : latest=5.1.6, local=5.1.6
tf2_tools                                 : latest=0.25.16, local=0.25.12
tango_icons_vendor                        : latest=0.1.1, local=0.1.1
dwb_msgs                                  : latest=1.1.18, local=1.1.18
ament_cmake                               : latest=1.3.12, local=1.3.11
diagnostic_common_diagnostics             : latest=4.0.6, local=4.0.3
nav2_waypoint_follower                    : latest=1.1.18, local=1.1.18
rclcpp_components                         : latest=16.0.14, local=16.0.12
launch_xml                                : latest=1.0.10, local=1.0.8
ros2component                             : latest=0.18.13, local=0.18.12
depth_image_proc                          : latest=3.0.8, local=3.0.7
rosidl_typesupport_fastrtps_cpp           : latest=2.2.2, local=2.2.2
costmap_queue                             : latest=1.1.18, local=1.1.18
ament_cmake_export_definitions            : latest=1.3.12, local=1.3.11
composition_interfaces                    : latest=1.2.2, local=1.2.1
nav2_voxel_grid                           : latest=1.1.18, local=1.1.18
nav2_bringup                              : latest=1.1.18, local=1.1.18
nav2_behavior_tree                        : latest=1.1.18, local=1.1.18
laser_geometry                            : latest=2.4.0, local=2.4.0
sensor_msgs                               : latest=4.9.0, local=4.8.0
self_test                                 : latest=4.0.6, local=4.0.3
rmw_cyclonedds_cpp                        : latest=1.3.4, local=1.3.4
rcl_interfaces                            : latest=1.2.2, local=1.2.1
image_transport                           : latest=3.1.12, local=3.1.11
image_transport_plugins                   : latest=2.5.3, local=2.5.3
robot_state_publisher                     : latest=3.0.3, local=3.0.3
image_publisher                           : latest=3.0.8, local=3.0.7
ros2param                                 : latest=0.18.13, local=0.18.12
smclib                                    : latest=4.1.2, local=4.1.2
ament_cmake_export_dependencies           : latest=1.3.12, local=1.3.11
tf2_py                                    : latest=0.25.16, local=0.25.12
ament_lint                                : latest=0.12.14, local=0.12.12
nav2_rotation_shim_controller             : latest=1.1.18, local=1.1.18
ament_index_cpp                           : latest=1.4.0, local=1.4.0
rclcpp_action                             : latest=16.0.14, local=16.0.12
nav2_costmap_2d                           : latest=1.1.18, local=1.1.18
nav2_util                                 : latest=1.1.18, local=1.1.18
rosidl_runtime_c                          : latest=3.1.7, local=3.1.6
ament_cmake_gmock                         : latest=1.3.12, local=1.3.11
tf2_eigen_kdl                             : latest=0.25.16, local=0.25.12
rqt_reconfigure                           : latest=1.1.2, local=1.1.2
rcpputils                                 : latest=2.4.5, local=2.4.5
rosbridge_suite                           : latest=2.0.1, local=2.0.0
nav2_bt_navigator                         : latest=1.1.18, local=1.1.18
ament_cmake_export_targets                : latest=1.3.12, local=1.3.11
ament_cmake_ros                           : latest=0.10.0, local=0.10.0
ament_index_python                        : latest=1.4.0, local=1.4.0
tf2_msgs                                  : latest=0.25.16, local=0.25.12
camera_calibration_parsers                : latest=3.1.12, local=3.1.11
python_orocos_kdl_vendor                  : latest=0.2.5, local=0.2.5
ament_cmake_flake8                        : latest=0.12.14, local=0.12.12
diagnostic_msgs                           : latest=4.9.0, local=4.8.0
rqt_gui                                   : latest=1.1.7, local=1.1.7
ros2bag                                   : latest=0.15.15, local=0.15.14
nav2_msgs                                 : latest=1.1.18, local=1.1.18
urdf                                      : latest=2.6.1, local=2.6.1
spdlog_vendor                             : latest=1.3.1, local=1.3.1
rosbag2_storage_default_plugins           : latest=0.15.15, local=0.15.14
tf2_eigen                                 : latest=0.25.16, local=0.25.12
rcl_yaml_param_parser                     : latest=5.3.10, local=5.3.9
rmw_implementation_cmake                  : latest=6.1.2, local=6.1.2
rosbag2                                   : latest=0.15.15, local=0.15.14
tf2_ros                                   : latest=0.25.16, local=0.25.12
ament_copyright                           : latest=0.12.14, local=0.12.12
rqt_gui_py                                : latest=1.1.7, local=1.1.7
tinyxml_vendor                            : latest=0.8.3, local=0.8.3
shared_queues_vendor                      : latest=0.15.15, local=0.15.14
geometry2                                 : latest=0.25.16, local=0.25.12
nav2_smac_planner                         : latest=1.1.18, local=1.1.18
diagnostic_aggregator                     : latest=4.0.6, local=4.0.3
rosidl_parser                             : latest=3.1.7, local=3.1.6
launch_testing_ros                        : latest=0.19.10, local=0.19.9
ament_cmake_export_link_flags             : latest=1.3.12, local=1.3.11
rosidl_runtime_cpp                        : latest=3.1.7, local=3.1.6
mcap_vendor                               : latest=0.15.15, local=0.15.14
ament_cmake_lint_cmake                    : latest=0.12.14, local=0.12.12
ament_xmllint                             : latest=0.12.14, local=0.12.12
eigen3_cmake_module                       : latest=0.1.1, local=0.1.1
interactive_markers                       : latest=2.3.2, local=2.3.2
nav2_rviz_plugins                         : latest=1.1.18, local=1.1.18
rosidl_default_runtime                    : latest=1.2.0, local=1.2.0
launch_yaml                               : latest=1.0.10, local=1.0.8
rosbag2_transport                         : latest=0.15.15, local=0.15.14
rosidl_typesupport_cpp                    : latest=2.0.2, local=2.0.2
sros2_cmake                               : latest=0.10.7, local=0.10.6
trajectory_msgs                           : latest=4.9.0, local=4.8.0
libcurl_vendor                            : latest=3.1.3, local=3.1.3
python_qt_binding                         : latest=1.1.2, local=1.1.2
tf2_ros_py                                : latest=0.25.16, local=0.25.12
rosbag2_py                                : latest=0.15.15, local=0.15.14
ros2service                               : latest=0.18.13, local=0.18.12
action_msgs                               : latest=1.2.2, local=1.2.1
rqt_graph                                 : latest=1.3.1, local=1.3.1
rviz_rendering                            : latest=11.2.19, local=11.2.16
resource_retriever                        : latest=3.1.3, local=3.1.3
actionlib_msgs                            : latest=4.9.0, local=4.8.0
ament_cppcheck                            : latest=0.12.14, local=0.12.12
rosbag2_storage_mcap                      : latest=0.15.15, local=0.15.14
rqt_console                               : latest=2.0.3, local=2.0.3
rosbag2_cpp                               : latest=0.15.15, local=0.15.14
nav2_amcl                                 : latest=1.1.18, local=1.1.18
rcl_lifecycle                             : latest=5.3.10, local=5.3.9
ompl                                      : latest=1.7.0, local=1.6.0
rosidl_typesupport_introspection_c        : latest=3.1.7, local=3.1.6
ros_workspace                             : latest=1.0.2, local=1.0.2
ament_cmake_auto                          : latest=1.3.12, local=1.3.11
rosapi_msgs                               : latest=2.0.1, local=2.0.0
stereo_image_proc                         : latest=3.0.8, local=3.0.7
diagnostics                               : latest=4.0.6, local=4.0.3
rosidl_typesupport_c                      : latest=2.0.2, local=2.0.2
rosx_introspection                        : latest=1.0.2, local=1.0.2
example_interfaces                        : latest=0.9.3, local=0.9.3
dwb_plugins                               : latest=1.1.18, local=1.1.18
std_msgs                                  : latest=4.9.0, local=4.8.0
ament_cmake_test                          : latest=1.3.12, local=1.3.11
qt_gui_py_common                          : latest=2.2.4, local=2.2.4
cv_bridge                                 : latest=3.2.1, local=3.2.1
orocos_kdl_vendor                         : latest=0.2.5, local=0.2.5
zstd_vendor                               : latest=0.15.15, local=0.15.14
tf2                                       : latest=0.25.16, local=0.25.12
stereo_msgs                               : latest=4.9.0, local=4.8.0
vision_msgs_rviz_plugins                  : latest=4.1.1, local=4.1.1
bondcpp                                   : latest=4.1.2, local=4.1.2
ros2doctor                                : latest=0.18.13, local=0.18.12
tf2_bullet                                : latest=0.25.16, local=0.25.12
fastrtps_cmake_module                     : latest=2.2.2, local=2.2.2
launch_testing                            : latest=1.0.10, local=1.0.8
rosbridge_msgs                            : latest=2.0.1, local=2.0.0
ros2multicast                             : latest=0.18.13, local=0.18.12
ament_cmake_gen_version_h                 : latest=1.3.12, local=1.3.11
rosbag2_storage                           : latest=0.15.15, local=0.15.14
tf2_kdl                                   : latest=0.25.16, local=0.25.12
statistics_msgs                           : latest=1.2.2, local=1.2.1
ament_cmake_copyright                     : latest=0.12.14, local=0.12.12
qt_dotgraph                               : latest=2.2.4, local=2.2.4
ros2node                                  : latest=0.18.13, local=0.18.12
demo_nodes_py                             : latest=0.20.5, local=0.20.5
rosidl_generator_c                        : latest=3.1.7, local=3.1.6
rosidl_typesupport_introspection_cpp      : latest=3.1.7, local=3.1.6
ament_cmake_version                       : latest=1.3.12, local=1.3.11
domain_coordinator                        : latest=0.10.0, local=0.10.0
rosbridge_library                         : latest=2.0.1, local=2.0.0
ros2lifecycle                             : latest=0.18.13, local=0.18.12
nav2_behaviors                            : latest=1.1.18, local=1.1.18
ament_cmake_export_libraries              : latest=1.3.12, local=1.3.11
rosidl_typesupport_fastrtps_c             : latest=2.2.2, local=2.2.2
std_srvs                                  : latest=4.9.0, local=4.8.0
launch                                    : latest=1.0.10, local=1.0.8
kdl_parser                                : latest=2.6.4, local=2.6.4
tf2_sensor_msgs                           : latest=0.25.16, local=0.25.12
rosapi                                    : latest=2.0.1, local=2.0.0
uncrustify_vendor                         : latest=2.0.2, local=2.0.2
ament_cmake_include_directories           : latest=1.3.12, local=1.3.11
rviz2                                     : latest=11.2.19, local=11.2.16
builtin_interfaces                        : latest=1.2.2, local=1.2.1
nav2_mppi_controller                      : latest=1.1.18, local=1.1.18
ament_lint_common                         : latest=0.12.14, local=0.12.12
keyboard_handler                          : latest=0.0.5, local=0.0.5
rmw                                       : latest=6.1.2, local=6.1.2
rosidl_generator_cpp                      : latest=3.1.7, local=3.1.6
rqt_py_common                             : latest=1.1.7, local=1.1.7
pybind11_vendor                           : latest=2.4.2, local=2.4.2
ros_core                                  : latest=0.10.0, local=0.10.0
rviz_default_plugins                      : latest=11.2.19, local=11.2.16
ament_uncrustify                          : latest=0.12.14, local=0.12.12
geometry_msgs                             : latest=4.9.0, local=4.8.0
rosidl_cli                                : latest=3.1.7, local=3.1.6
ament_cmake_pytest                        : latest=1.3.12, local=1.3.11
rpyutils                                  : latest=0.2.1, local=0.2.1
class_loader                              : latest=2.2.0, local=2.2.0
nav_2d_msgs                               : latest=1.1.18, local=1.1.18
tinyxml2_vendor                           : latest=0.7.6, local=0.7.6
rcl_action                                : latest=5.3.10, local=5.3.9
nav_msgs                                  : latest=4.9.0, local=4.8.0
rclcpp_lifecycle                          : latest=16.0.14, local=16.0.12
nav2_core                                 : latest=1.1.18, local=1.1.18
ament_cpplint                             : latest=0.12.14, local=0.12.12
ros2action                                : latest=0.18.13, local=0.18.12
nav2_theta_star_planner                   : latest=1.1.18, local=1.1.18
common_interfaces                         : latest=4.9.0, local=4.8.0
foxglove_msgs                             : latest=2.3.0, local=2.3.0
isaac_ros_stereo_image_proc               : latest=, local=3.2.5
isaac_ros_managed_nitros                  : latest=, local=3.2.5
isaac_ros_nitros                          : latest=, local=3.2.5
isaac_ros_tensor_rt                       : latest=, local=3.2.10
gxf_isaac_gems                            : latest=, local=3.2.5
gxf_isaac_utils                           : latest=, local=3.2.5
gxf_isaac_sight                           : latest=, local=3.2.5
gxf_isaac_message_compositor              : latest=, local=3.2.5
gxf_isaac_tensor_rt                       : latest=, local=3.2.5
isaac_ros_image_proc                      : latest=, local=3.2.10
isaac_ros_nitros_camera_info_type         : latest=, local=3.2.5
gxf_isaac_argus                           : latest=, local=3.2.11
gxf_isaac_sgm                             : latest=, local=3.2.5
isaac_ros_tensor_proc                     : latest=, local=3.2.10
isaac_ros_image_pipeline                  : latest=, local=3.2.5
isaac_ros_common                          : latest=, local=3.2.5
gxf_isaac_ros_messages                    : latest=, local=3.2.5
isaac_ros_argus_camera                    : latest=, local=3.2.5
gxf_isaac_camera_utils                    : latest=, local=3.2.5
isaac_ros_gxf                             : latest=, local=3.2.5
isaac_ros_nitros_image_type               : latest=, local=3.2.5
isaac_ros_depth_image_proc                : latest=, local=3.2.5
isaac_ros_nitros_point_cloud_type         : latest=, local=3.2.5
isaac_ros_nitros_tensor_list_type         : latest=, local=3.2.5
gxf_isaac_tensorops                       : latest=, local=3.2.5
gxf_isaac_image_flip                      : latest=, local=3.2.5
gxf_isaac_messages                        : latest=, local=3.2.5
gxf_isaac_optimizer                       : latest=, local=3.2.10
gxf_isaac_point_cloud                     : latest=, local=3.2.5
gxf_isaac_depth_image_proc                : latest=, local=3.2.5
gxf_isaac_atlas                           : latest=, local=3.2.5
isaac_ros_dnn_image_encoder               : latest=, local=3.2.10
gxf_isaac_rectify                         : latest=, local=3.2.5
gxf_isaac_timestamp_correlator            : latest=, local=3.2.5
gxf_isaac_gxf_helpers                     : latest=, local=3.2.5
isaac_ros_nitros_disparity_image_type     : latest=, local=3.2.5
isaac_ros_tensor_list_interfaces          : latest=, local=3.2.5
moveit2_tutorials                         : latest=, local=0.1.0
moveit_hybrid_planning                    : latest=2.5.9, local=2.5.9
rviz_marker_tools                         : latest=, local=0.1.3
moveit_task_constructor_capabilities      : latest=, local=0.1.3
moveit_task_constructor_core              : latest=, local=0.1.3
moveit_task_constructor_msgs              : latest=, local=0.1.3
moveit_task_constructor_demo              : latest=, local=0.1.3
moveit_task_constructor_visualization     : latest=, local=0.1.3
moveit_resources_panda_description        : latest=2.0.7, local=2.0.7
moveit_resources_panda_moveit_config      : latest=2.0.7, local=2.0.7
moveit_resources                          : latest=2.0.7, local=2.0.7
moveit_resources_fanuc_description        : latest=2.0.7, local=2.0.7
moveit_resources_pr2_description          : latest=2.0.7, local=2.0.7
moveit_resources_fanuc_moveit_config      : latest=2.0.7, local=2.0.7
py_binding_tools                          : latest=2.0.1, local=2.0.1
moveit_visual_tools                       : latest=4.1.2, local=4.1.2
object_recognition_msgs                   : latest=2.0.0, local=2.0.0
sdl2_vendor                               : latest=3.3.0, local=3.3.0
moveit_setup_core_plugins                 : latest=2.5.9, local=2.5.9
pose_broadcaster                          : latest=2.49.1, local=2.44.0
octomap_msgs                              : latest=2.0.1, local=2.0.1
moveit_ros                                : latest=2.5.9, local=2.5.9
generate_parameter_library_py             : latest=0.5.0, local=0.4.0
force_torque_sensor_broadcaster           : latest=2.49.1, local=2.44.0
moveit_ros_occupancy_map_monitor          : latest=2.5.9, local=2.5.9
transmission_interface                    : latest=2.51.0, local=2.50.0
gripper_controllers                       : latest=2.49.1, local=2.44.0
eigen_stl_containers                      : latest=1.1.0, local=1.1.0
octomap                                   : latest=1.9.8, local=1.9.8
warehouse_ros_sqlite                      : latest=1.0.5, local=1.0.5
ur                                        : latest=2.8.1, local=2.7.0
moveit_msgs                               : latest=2.2.1, local=2.2.1
xacro                                     : latest=2.0.13, local=2.0.13
ur_dashboard_msgs                         : latest=2.8.1, local=2.7.0
geometric_shapes                          : latest=2.3.2, local=2.3.2
filters                                   : latest=2.2.2, local=2.1.0
moveit_setup_controllers                  : latest=2.5.9, local=2.5.9
moveit_ros_benchmarks                     : latest=2.5.9, local=2.5.9
ros2_controllers_test_nodes               : latest=2.49.1, local=2.44.0
forward_command_controller                : latest=2.49.1, local=2.44.0
joint_trajectory_controller               : latest=2.49.1, local=2.44.0
ur_msgs                                   : latest=2.3.0, local=2.2.0
moveit_ros_robot_interaction              : latest=2.5.9, local=2.5.9
moveit_setup_assistant                    : latest=2.5.9, local=2.5.9
moveit_simple_controller_manager          : latest=2.5.9, local=2.5.9
warehouse_ros                             : latest=2.0.5, local=2.0.5
ur_controllers                            : latest=2.8.1, local=2.7.0
moveit_ros_move_group                     : latest=2.5.9, local=2.5.9
ros2_control                              : latest=2.51.0, local=2.50.0
moveit_setup_srdf_plugins                 : latest=2.5.9, local=2.5.9
graph_msgs                                : latest=0.2.0, local=0.2.0
position_controllers                      : latest=2.49.1, local=2.44.0
moveit_setup_app_plugins                  : latest=2.5.9, local=2.5.9
random_numbers                            : latest=2.0.1, local=2.0.1
moveit_ros_perception                     : latest=2.5.9, local=2.5.9
srdfdom                                   : latest=2.0.7, local=2.0.7
urdfdom_py                                : latest=1.2.1, local=1.2.1
parameter_traits                          : latest=0.5.0, local=0.4.0
rviz_visual_tools                         : latest=4.1.4, local=4.1.4
moveit_planners                           : latest=2.5.9, local=2.5.9
moveit_ros_visualization                  : latest=2.5.9, local=2.5.9
topic_based_ros2_control                  : latest=0.2.0, local=0.2.0
ur_client_library                         : latest=2.2.0, local=1.9.0
controller_manager_msgs                   : latest=2.51.0, local=2.50.0
chomp_motion_planner                      : latest=2.5.9, local=2.5.9
pilz_industrial_motion_planner            : latest=2.5.9, local=2.5.9
ur_moveit_config                          : latest=2.8.1, local=2.7.0
moveit_planners_chomp                     : latest=2.5.9, local=2.5.9
tcb_span                                  : latest=1.0.2, local=1.0.2
controller_manager                        : latest=2.51.0, local=2.50.0
ur_calibration                            : latest=2.8.1, local=2.7.0
moveit_servo                              : latest=2.5.9, local=2.5.9
hardware_interface                        : latest=2.51.0, local=2.50.0
control_msgs                              : latest=4.8.0, local=4.8.0
ur_robot_driver                           : latest=2.8.1, local=2.7.0
moveit_ros_planning_interface             : latest=2.5.9, local=2.5.9
moveit_configs_utils                      : latest=2.5.9, local=2.5.9
realtime_tools                            : latest=2.14.0, local=2.13.0
generate_parameter_library                : latest=0.5.0, local=0.4.0
joint_state_publisher                     : latest=2.4.0, local=2.4.0
moveit_planners_ompl                      : latest=2.5.9, local=2.5.9
moveit_ros_planning                       : latest=2.5.9, local=2.5.9
moveit                                    : latest=2.5.9, local=2.5.9
tl_expected                               : latest=1.0.2, local=1.0.2
moveit_ros_warehouse                      : latest=2.5.9, local=2.5.9
moveit_setup_framework                    : latest=2.5.9, local=2.5.9
control_toolbox                           : latest=3.6.1, local=3.6.1
moveit_plugins                            : latest=2.5.9, local=2.5.9
ur_description                            : latest=2.6.0, local=2.5.0
controller_interface                      : latest=2.51.0, local=2.50.0
joint_state_publisher_gui                 : latest=2.4.0, local=2.4.0
joint_limits                              : latest=2.51.0, local=2.50.0
moveit_kinematics                         : latest=2.5.9, local=2.5.9
velocity_controllers                      : latest=2.49.1, local=2.44.0
ros2controlcli                            : latest=2.51.0, local=2.50.0
joint_state_broadcaster                   : latest=2.49.1, local=2.44.0
moveit_core                               : latest=2.5.9, local=2.5.9
joy                                       : latest=3.3.0, local=3.3.0
moveit_common                             : latest=2.5.9, local=2.5.9
launch_param_builder                      : latest=0.1.1, local=0.1.1
ros2_control_test_assets                  : latest=2.51.0, local=2.50.0
ur_bringup                                : latest=2.8.1, local=2.7.0
ament_cmake_google_benchmark              : latest=1.3.12, local=1.3.11
mimick_vendor                             : latest=0.2.8, local=0.2.8
test_msgs                                 : latest=1.2.2, local=1.2.1
performance_test_fixture                  : latest=0.0.9, local=0.0.9
negotiated                                : latest=, local=0.1.0
negotiated_interfaces                     : latest=, local=0.1.0

   PLATFORM INFORMATION
system           : Linux
platform info    : Linux-5.15.148-tegra-aarch64-with-glibc2.35
release          : 5.15.148-tegra
processor        : aarch64

   QOS COMPATIBILITY LIST
compatibility status    : No publisher/subscriber pairs found

   RMW MIDDLEWARE
middleware name    : rmw_fastrtps_cpp

   ROS 2 INFORMATION
distribution name      : humble
distribution type      : ros2
distribution status    : active
release platforms      : {'rhel': ['8'], 'ubuntu': ['jammy']}

   TOPIC LIST
topic               : none
publisher count     : 0
subscriber count    : 0
```
</details>


### Steps to reproduce issue

To reproduce, launch node `my_node` with params `global_param` and `specific_param` using the YAML file below:

```
/**:
  ros__parameters:
    global_param: 1
  my_node:
    ros__parameters:
      specific_param: 2
```

However, using the following YAML file works correctly:


```
/**:
  my_node:
    ros__parameters:
      specific_param: 2
  ros__parameters:
    global_param: 1
```


### Expected behavior

The `specific_param` is applied to `my_node` and `global_param` is applied to all nodes. This happens when using the second file.

### Actual behavior

With the first file i get the error:
```
[ERROR] [1755072385.906647784] [rcl]: Failed to parse global arguments
[my_node-1]  Traceback (most recent call last):
[my_node-1]    File "/myrobot/install/my_package/lib/my_package/my_node", line 33, in <module>
[my_node-1]      sys.exit(load_entry_point('my_package', 'console_scripts', 'my_node')())
[my_node-1]    File "/myrobot/build/my_package/my_package/my_node.py", line 112, in main
[my_node-1]      rclpy.init(args=args)
[my_node-1]    File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 88, in init
[my_node-1]      context.init(args, domain_id=domain_id)
[my_node-1]    File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/context.py", line 72, in init
[my_node-1]      self.__context = _rclpy.Context(
[my_node-1]  rclpy._rclpy_pybind11.RCLError: failed to initialize rcl: Couldn't parse params file: '--params-file config/config.yaml'. Error: Internal error creating param namespace at line 6, at ./src/parse.c:674, at ./src/rcl/arguments.c:406

```

### Additional information

_No response_

--- a/rcl_yaml_param_parser/src/parse.c
+++ b/rcl_yaml_param_parser/src/parse.c
@@ -664,12 +664,6 @@ rcutils_ret_t parse_key(
             break;
           }
 
-          ret = rem_name_from_ns(ns_tracker, NS_TYPE_NODE, allocator);
-          if (RCUTILS_RET_OK != ret) {
-            RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
-              "Internal error adding node namespace at line %d", line_num);
-            break;
-          }
           /// Bump the map level to PARAMS
           (*map_level)++;
         }
@@ -855,9 +849,7 @@ rcutils_ret_t parse_file_events(
         is_new_map = true;
         is_key = true;
         /// Disable new map if it is PARAMS_KEY map
-        if ((MAP_PARAMS_LVL == map_level) &&
-          ((map_depth - (ns_tracker->num_node_ns + 1U)) == 2U))
-        {
+        if ((MAP_PARAMS_LVL == map_level) && ((map_depth - ns_tracker->num_node_ns) == 2U)) {
           is_new_map = false;
         }
         break;

--- a/rcl_yaml_param_parser/test/wildcards.yaml
+++ b/rcl_yaml_param_parser/test/wildcards.yaml
@@ -27,6 +27,34 @@
     ros__parameters:
       id: 10
       name: "front lidar"
+/**:
+  ros__parameters:
+    global_param: 1
+  some_node5:
+    ros__parameters:
+      id: 10
+      name: "front lidar"
+/**:
+  some_node6:
+    ros__parameters:
+      id: 10
+      name: "front lidar"
+  ros__parameters:
+    global_param: 1
+/*:
+  ros__parameters:
+    global_param: 1
+  some_node7:
+    ros__parameters:
+      id: 10
+      name: "front lidar"
+/*:
+  some_node8:
+    ros__parameters:
+      id: 10
+      name: "front lidar"
+  ros__parameters:
+    global_param: 1
 /foo1/**:
   ros__parameters:
     id: 10

